### DISCUSSION TITLE
Widget, CSS for dark mode no more respected on widget but works on the pop up

### DISCUSSION THREAD

**Discussion by PeopleInside** [LiveHelperChat/livehelperchat#2120] [General]
Hi, I noticed today that the CSS code that I was using in the widget to activate a dark mode theme when OS system was set on dark is no more working on widget but still work on pop up also if there is no CSS dedicated for pop relative to dark mode.

Maybe some update has broken something or changed CSS priority?

I tried to update to the last commit https://github.com/LiveHelperChat/livehelperchat/commit/81e45e5ae36e4dd6363dd4a32c8f98130c8191f3 but still have the issue also cleaning cache.

I had never changed LHC settings so I think the cause can be an update as my custom Dark mode CSS still be there and also is working for the pop up.

![Screenshot 2024-07-15 210839](https://github.com/user-attachments/assets/7e80ba62-a279-4e44-a848-88f8cb465694)
![Screenshot 2024-07-15 211040](https://github.com/user-attachments/assets/56b2de28-deea-4a5b-9dad-b8699abbbe58)
![Screenshot 2024-07-15 210858](https://github.com/user-attachments/assets/52c2d991-fad3-4d08-a278-fbcc87e6a88e)
![Screenshot 2024-07-15 211056](https://github.com/user-attachments/assets/d42c1fbf-c990-4822-8af1-2de548a6b5b0)
![Screenshot 2024-07-15 211254](https://github.com/user-attachments/assets/79a79713-a003-4ec1-a324-022f37732d6b)

**Comment by PeopleInside** [comment]
If is not clear in the screenshots I show that the widget has no more a dark theme but is a light one also if custom.
In the pop up all works as expected as in the past but also widget in the past was working with my custom dark mode CSS.
Now is not working.

**Reply by remdex** [reply]
Status widget is now web component now, so you will have to adjust your theme css.

**Reply by PeopleInside** [reply]
Thanks for this information.
What means is a web component and I need adjust my theme css?
From the browser console I can see there is the same css elements mmm the css code still be at the same field or I need move the css?

**Reply by PeopleInside** [reply]
My CSS code work only on admin side, no more on visitor side:
![immagine](https://github.com/user-attachments/assets/5cf34086-c79c-4ae3-949b-29ccc36e0c5d)

What I have to do?

As you can see in the screenshot visitor widget preview I have the dark mode on the widget but on visitor side the widget is no more dark. My CSS code seems to be correct but are no more applied.

**Reply by remdex** [reply]
Please educate yourself about webcomponetns and their styling...

**Reply by PeopleInside** [reply]
> Please educate yourself about webcomponetns and their styling...

What I do wrong?
My CSS was working, is not working anymore after the LHC update.
Now it works only on admin preview no more on visitor side and I dont understand why.

![lhc](https://github.com/user-attachments/assets/782f58a9-d33f-4513-8195-665d625d1fc4)

As you can see in the admin widget preview my old code works, is no more working on the website visitor side.

**Comment by remdex** [comment]
These rules works on my env if are put in `Status widget additional CSS`. Color changes only if you change windows color theme. They did not reacted to chrome color theme.

```
@media (prefers-color-scheme: dark) {
  #lhc_status_container #status-icon{
    background-color:red!important;
    }
}


@media (prefers-color-scheme: light) {
  #lhc_status_container #status-icon{
    background-color:blue!important;
    }
}
```

**Reply by PeopleInside** [reply]
Hum...
My CSS code to apply the dark mode seems correct.
I have already `@media (prefers-color-scheme: dark)` rules.

What i see today is that if WIndows OS change color from dark then seems on Firefox the LHC chat widget seems never activate the dark mode. Seems if I simulate in Firefox with the developer tool the dark ... the widget dark mode works.

It's strange I have to investigate this maybe be sure is not an issue just with Firefox. It's a new issue, I never had this issue in the past.

**Reply by PeopleInside** [reply]
I can confirm, my code works but seems Firefox is not propriely apply to LHC if Windows color change.
It's a strange issue, I will try to open a BUG in Firefox but looks strange it's a Firefox bug.
Very strange :) with Chrome no issues.

https://github.com/webcompat/web-bugs/issues/139227

**Comment by PeopleInside** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1908060

**Comment by PeopleInside** [comment]
Hi @remdex,
seems in the widget iframe there is `color-scheme: light` and I belive i dont added this. IS maybe some update of LHC?
The firefox bug has been closed as not bug in Firefox but an issue in LHC widget
https://bugzilla.mozilla.org/show_bug.cgi?id=1908060#c5

I checked and in my CSS widget code I have just color scheme dark, i haven't put light scheme.
I believe at this point some widget update has introduced a light CSS that seems is causing issue with the CSS dark mode I inserted and was working before LHC update.

Firefox has closed the bug report to not a Firefox bug.
Chrome has a bug that show the widget dark but something in the LHC widget seems told to use the light schema from what I'm understanding. I have no light schema CSS rules so if this exist seems is from LHC core CSS code.

Firefo reply:

`The <iframe> explicitly has color-scheme: light so it's expected. `

LHC widget is set to be always light? This is an updated made by you and if I want have the dark theme?
In the past my CSS dark mode code was working on LHC widget. I never edited my CSS dark code so I suppose is an update of LHC that is causing the issue.

![Screenshot 2024-07-22 112250](https://github.com/user-attachments/assets/dd2cb1c9-6247-4f31-bfae-f83a3c1d7256)

**Reply by PeopleInside** [reply]
To fix the issue caused by the CSS light rule inserted not by me but seems is in the core I need add additional CSS to unset the color scheme in the field Custom page CSS (new widget only):

![immagine](https://github.com/user-attachments/assets/a2595191-88fd-4211-a725-d68c5509df9b)
